CODECONVENTIONS.md

```diff
@@ -229,6 +229,25 @@ Names:
 - Use underscore_case, not camelCase for all names.
 - Use CAPS_WITH_UNDERSCORE for enums and macros.
 - When defining a type use underscore_case and put '_t' after it.
+- Non-static functions and variables declared in a header and defined in a
+  source file should generally share a common prefix. Usually the prefix matches
+  the file name (i.e. items defined in `py/obj.c` are declared in `py/obj.h` and
+  should be prefixed `mp_obj_`). There are exceptions, for example where one
+  header file contains declarations implemented in multiple source files for
+  expediency.
+
+Static names:
+- For static functions and variables exposed to Python (i.e. a static C function
+  that is wrapped in `MP_DEFINE_CONST_FUN_...` and attached to a module), use
+  the file-level shared common prefix, i.e. name them as if the function or
+  variable was not static.
```

Comment:
I'm not sure about this point.  There are quite a few examples that don't follow this, or follow a variation of it.  (Or maybe I don't understand exactly what you wrote?)

Eg the `marshal` module has static `marshal_loads()` and `marshal_dumps()` that are exposed to Python.  But if they were non-static I think they'd be called `mp_marshal_loads()` etc.  Similarly the `time` module has static functions like `time_time()` and `time_time_ns()`.

A different example: `openamp.Endpoint` has static functions like `endpoint_send()` which could probably be `openamp_endpoint_send()`.

Maybe all that really needs changing in this paragraph is to say that such static functions don't have the `mp_` prefix?

Reply:
Thanks for taking a look, Damien. This PR is me trying to document the most commonly used existing practices as a baseline, so I'm definitely happy to change it to what you think is most accurate here.

Related question: Should all non-static identifiers have an `mp_` prefix, as a general rule?

Reply:
Unfortunately I don't think existing code is consistent enough to have a general rule (for static functions).  There will be lots of exceptions to whatever rule is written down here.

Considering @dlech point below that everything should be prefixed with `mp_` (which I tend to agree with, makes it simpler), maybe we can just keep the text you've written.

> Related question: Should all non-static identifiers have an `mp_` prefix, as a general rule?

Yes, I think so.  At least in `py/` and `extmod/`.  For ports... they are less consistent.

Reply:
I've updated based on these comments and our discussion just now.

Reply:
Made another small update